android/src/main/java/com/mr/flutter/plugin/filepicker/FileUtils.java

```diff
@@ -92,28 +92,27 @@ public static String getFileName(Uri uri, final Context context) {
     }
 
 
-    public static Uri compressImage(Uri originalImageUri, int compressionQuality,Context context) {
-        String originalImagePath = getRealPathFromURI(context,originalImageUri);
-       Uri compressedUri=null;
-       File compressedFile=null;
+    public static Uri compressImage(Uri originalImageUri, int compressionQuality, Context context) {
+        Uri compressedUri;
         try {
-             compressedFile=createImageFile();
-            Bitmap originalBitmap = BitmapFactory.decodeFile(originalImagePath);
-            String file_path = Environment.getExternalStorageDirectory().getAbsolutePath() +
-                    "/FilePicker";
+            File compressedFile = createImageFile();
+            Bitmap originalBitmap = BitmapFactory.decodeStream(context.getContentResolver().openInputStream(originalImageUri));
```

Comment:
We probably need to close the input stream once we're done ith it?

Reply:
Of course, sorry for the oversight.